### PR TITLE
Added documentation for (part of) the Quantum Mechanics module.

### PR DESCRIPTION
**Pull request by rpmuller**
Hey guys, here's some text for the QM section. There's also an image in the images directory of a quantum circuit example.

### CODE REVIEW

**Comment by asmeurer** [comment]
I think you forgot to add the image.

**Comment by rpmuller** [comment]
Can someone give me instructions on how to add a file to an existing pull request? Do I first add it to the previous commit, then re-push to the forked branch, and then create a new pull request?

**Comment by certik** [comment]
@rpmuller just go to the branch you sent the pull request from, in your case `master` (next time you should use a separate branch, but that's fine for now), and just create a new commit that adds the image. Then push that new commit to your fork (`git push origin master`, assuming `origin` points to your fork). Then the new commit will automatically appear in this PR.

**Comment by rpmuller** [comment]
Done. See if that worked.

**Comment by certik** [comment]
@rpmuller it worked! Thanks for writing this up, I appreciate it.

**Comment by rpmuller** [comment]
@certik My pleasure.

**Comment by asmeurer** [comment]
Can circuit plot generate vector images? That would look better than a png.

**Comment by rpmuller** [comment]
@asmeurer Agreed. I bet it can, since it's just matplotlib. Let me think about it and figure something out.

File: domain_specific.tex [hunk 1]
```diff
@@ -61,8 +61,70 @@ \subsubsection{Mechanics}
...
+\end{center}
+\end{figure}
+
+Finally, there are also capabilities to plot circuit diagrams for simple circuits.
+\begin{verbatim}
+>>> from sympy.physics.quantum.circuitplot import CircuitPlot
+>>> from sympy.physics.quantum.gate import H, CNOT
+>>> CircuitPlot(CNOT(1,0)*H(1),2)
+\end{verbatim}
+This results in figure \ref{fig-circuitplot-example}.
```
Comment (asmeurer) [inline domain_specific.tex]: Use `~\ref`

Comment (asmeurer) [inline domain_specific.tex]: Also should "figure" be capitalized? I'm not sure what the convention is.

**Comment by asmeurer** [comment]
The text looks good. Just those comments.

**Comment by asmeurer** [comment]
This seems to work

``` py
CircuitPlot(CNOT(1,0)*H(1),2)._figure.savefig('circuit.pdf', format='pdf')
```

(you can also use eps, I'm not sure if that's better)

**Comment by rpmuller** [comment]
@asmeurer Made these changes. Used PDF format, capitalized "Figure", added the ~. Let me know if there's anything else.

**Comment by certik** [comment]
Thanks @rpmuller!

**Pull request merged**